src/pass/llvmgen.cpp

```diff
@@ -507,6 +572,54 @@ llvm::Value* LLVMGen::visit(Sym old_sym)
     return var;
 }
 
+void LLVMGen::init_cuda()
+{
+    InitializeAllTargets();
+    InitializeAllTargetMCs();
+    InitializeAllAsmPrinters();
+    InitializeAllAsmParsers();
+
+    llmod()->setTargetTriple("nvptx64-nvidia-cuda");
+}
+
+TargetMachine* LLVMGen::get_target()
+{
+    std::string Error;
+    const llvm::Target* Target =
+        llvm::TargetRegistry::lookupTarget("nvptx64-nvidia-cuda", Error);
+
+    llvm::TargetOptions opt;
+    llvm::TargetMachine* TM =
+        Target->createTargetMachine("nvptx64-nvidia-cuda", "sm_52",
+                                    "+ptx76",  // PTX version
+                                    opt, llvm::Reloc::Static);
+
+    llmod()->setDataLayout(TM->createDataLayout());
+
+    return TM;
+}
+
+string LLVMGen::BuildPTX(shared_ptr<Func> func, Module& llmod)
+{
+    LLVMGenCtx ctx(func);
+    LLVMGen llgen(ctx, llmod);
+    llgen.init_cuda();
```

Comment:
you can take a look how the CPU codegen and engine is built. there is also an initialization for CPU backend and it is written inside the Get() function of ExecEngine.